pip/commands/install.py

```diff
@@ -251,12 +251,12 @@ def run(self, options, args):
                     progress_bar=options.progress_bar,
                 )
 
-                self.populate_requirement_set(
-                    requirement_set, args, options, finder, session, self.name,
-                    wheel_cache
-                )
-
                 try:
+                    self.populate_requirement_set(
```

Comment:
Yes, it's still the `WheelBuilder` a few lines below that builds wheels for the requirement set. The crucial difference is that previously it only built wheels if they could be cached (e.g. if we've downloaded an sdist, so the wheel corresponds to a particular version). Now it also builds ephemeral wheels that won't be cached, e.g. if you install from a local directory.